disco/preprocess/select_timepoints2.py

```diff
@@ -166,166 +322,241 @@ def get_profile_data():
             the keys are power conversion asset categories: "demand" and "generation"
             the values are a list of paths pointing to the corresponding power conversions assets' .dss files such as "Loads.dss" and "PVSystems.dss" files
             example: category_path_dict = {'demand': [LOAD_PATH, EVLOAD_PATH], 'generation': [PV_PATH]}
         profile_files: a list of paths pointing to shape (profile) files such as "LoadShapes.dss"
     OUTPUT:
-        bus_data: 
+        bus_data:
         profile_data:
         ag_series:
-        head_time_indices: list of critical time indices when only feeder-head timeseries are considered 
+        head_time_indices: list of critical time indices when only feeder-head timeseries are considered
         critical_time_indices: all critical time indices (individual buses as well as feeder-head considered)
         compression_rate: ratio between the number of critical timepoints and the total number of timepoints in the timeseries
-                             
+
     """
-    folder, _ = os.path.split(path_to_master)
-    if destination_dir==None:
-        folder, _ = os.path.split(path_to_master)
-        destination_dir = folder
-    else:
-        os.makedirs(destination_dir, exist_ok=True)
-    
-    bus_data={}
-    load_feeder(path_to_master)
-    for category, param_classes in category_class_dict.items():
+    if not destination_dir.exists():
+        raise FileNotFoundError(f"destination_dir={destination_dir} does not exist")
+
+    if check_power_flow and not create_new_circuit:
+        logger.warning("Ignore check_power_flow because create_new_circuit is False")
+        check_power_flow = False
+
+    bus_data = {}
+    load_feeder(path_to_master, destination_dir, fix_master_file)
+    for category, param_classes in categories.items():
         for param_class in param_classes:
             bus_data = get_param_values(param_class, bus_data, category)
     profile_data = get_profile_data()
-    ag_series, head_time_indices, critical_time_indices, compression_rate = \
-        agregate_series(bus_data, 
-                        profile_data, 
-                        critical_conditions, 
-                        recreate_profiles, 
-                        destination_dir)
-    return bus_data, profile_data, ag_series, head_time_indices, critical_time_indices, compression_rate
-    
+    ag_series, head_time_indices, critical_time_indices, compression_rate = aggregate_series(
+        bus_data,
+        profile_data,
+        critical_conditions,
+        recreate_profiles,
+        destination_dir,
+        create_new_circuit,
+        check_power_flow,
+    )
+
+    logger.info("head_time_indices = %s length = %s", head_time_indices, len(head_time_indices))
+    logger.info(
+        "critical_time_indices = %s length = %s", critical_time_indices, len(critical_time_indices)
+    )
+    with open(destination_dir / "metadata.csv", "w") as f_out:
```

Comment:
@ann-sherin Need your input on what you want here. If you prefer, you can write the code to collect what you want.